### PR TITLE
refactor(freespace_planner): parameterize margin.

### LINKED ISSUE #1165: Add vehicle margin to parameter in freespace planner
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I've agreed with the maintainers that I can plan this task.

### Description

Add vehicle margin variable in freespace_planner as parameter to replace the constant variable in code.

### Purpose

Currently vehicle margin is a constant in freespace planner, but large margin like 1 meter in code  give less freespace to use often leads no solution and frequently replan according to our real parking test  result.  It's reasonable margin is flexible in parking scenario, user can tune their value in different scene.

### Possible approaches

add margin in parameter file and  read them in code.

### Definition of done

margin variable  in parameter file.

### PR DESCRIPTION
**Pull request by klu17**
## Description

Resolves #1165 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by shmpwk** [comment]
Thank you for your contribution! 
Could you also add config parameter [here](https://github.com/autowarefoundation/autoware.universe/blob/main/launch/tier4_planning_launch/config/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml) for upper launching layer?

**Comment by klu17** [comment]
Yes, I can do that! @shmpwk

**Comment by klu17** [comment]
@shmpwk Should I create a new issue?

**Comment by shmpwk** [comment]
@klu17 
https://github.com/autowarefoundation/autoware.universe/issues/1165 is enough, thank you.

**Comment by shmpwk** [comment]
@klu17 
Thank you for your refactoring.
I'm checking it works correctly. 
(But for me, it looks like it doesn't change anything. Let's me check for a while.)

Also, it is very helpful if you modify readme https://github.com/autowarefoundation/autoware.universe/tree/main/planning/freespace_planner

**Comment by shmpwk** [comment]
Thank you for adding readme.

Also thank you for summing up the commit, but you don't have to since actually we will squash marging the commit.

(And I'm still testing now......)

**Review by shmpwk — CHANGES_REQUESTED** [review]
please add `double margin;` for `NodeParam` here. 
https://github.com/autowarefoundation/autoware.universe/blob/main/planning/freespace_planner/include/freespace_planner/freespace_planner_node.hpp#L85-L86

**Review by shmpwk — CHANGES_REQUESTED** [review]
Could you declare parameter first?
```
p.margin_m = declare_parameter("margin_m", 1.0);
```

btw https://github.com/autowarefoundation/autoware.universe/blob/d27e83d671084e1714262d185dea51600a09add5/planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp#L223-L224

File: planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp
Comment (shmpwk) [inline planning/freespace_planner/src/freespace_planner/freespace_planner_node.cpp:468]:
```suggestion
  const double margin = node_param_.margin;
  extended_vehicle_shape.length += margin;
  extended_vehicle_shape.width += margin;
  extended_vehicle_shape.base2back += margin / 2;
```

**Review by shmpwk — CHANGES_REQUESTED** [review]

**Comment by shmpwk** [comment]
@klu17 Could you fix this?
https://github.com/autowarefoundation/autoware.universe/pull/1190#pullrequestreview-1022837633

**Comment by klu17** [comment]
Should be up now.

**Review by shmpwk — APPROVED** [review]
Nice! Thank you!
I tested these changes with planning simulator, and it works fine!
Please wait for the ci results and at least one more approval.

File: launch/tier4_planning_launch/config/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml
Comment (kenji-miyake) [inline launch/tier4_planning_launch/config/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml:11]:
```suggestion
    vehicle_shape_margin_m: 1.0
```

```suggestion
    vehicle_footprint_margin_m: 1.0
```

**Review by kenji-miyake — APPROVED** [review]
LGTM, but it's better to add some prefix to the parameter name.

File: launch/tier4_planning_launch/config/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml
Comment (shmpwk) [reply launch/tier4_planning_launch/config/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml:11]: IMO, `vehicle_shape_margin_m` is better rather than `vehicle_footprint_margin_m`.
@klu17 Could you also fix this? :bow:

Comment (klu17) [reply launch/tier4_planning_launch/config/scenario_planning/parking/freespace_planner/freespace_planner.param.yaml:11]: I'll go ahead and change it to `vehicle_footprint_margin_m`.

**Comment by shmpwk** [comment]
@klu17 Thank you for your full commitment!

**Pull request merged**
